posts/Yoann/2019-11-08-hibernate-search-quarkus.adoc

```diff
@@ -0,0 +1,1428 @@
+
+=== Initializing the project
+
+Quarkus provides a Maven plugin to initialize the layout of a new project.
+Just run this:
+
+[source]
+----
+$ mvn io.quarkus:quarkus-maven-plugin:0.28.0:create \
+    -DprojectGroupId=org.hibernate.demos \
+    -DprojectArtifactId=hsearch-quarkus \
+    -DclassName="org.hibernate.demos.quarkus.ClientResource" \
+    -Dpath="/client" \
+    -Dextensions="hibernate-orm-panache, hibernate-search-elasticsearch, resteasy-jsonb, jdbc-postgresql"
+----
+
+Then move to the created directory:
+
+[source]
+----
+$ cd hsearch-quarkus
+----
+
+The directory will contain everything you need to start coding right away:
+
+[source]
+----
+$ tree .
+hsearch-quarkus
+├── mvnw
+├── mvnw.cmd
+├── pom.xml
+└── src
+    ├── main
+    │   ├── docker
+    │   │   ├── Dockerfile.jvm
+    │   │   └── Dockerfile.native
+    │   ├── java
+    │   │   └── org
+    │   │       └── hibernate
+    │   │           └── demos
+    │   │               └── quarkus
+    │   │                   └── ClientResource.java
+    │   └── resources
+    │       ├── application.properties
+    │       └── META-INF
+    │           └── resources
+    │               └── index.html
+    └── test
+        └── java
+            └── org
+                └── hibernate
+                    └── demos
+                        └── quarkus
+                            ├── ClientResourceTest.java
+                            └── NativeClientResourceIT.java
+
+17 directories, 10 files
+----
+
+=== Tests
+
+The tests created by Quarkus are fine,
+and in a real-world application we should update them as we add more features to our applications.
+
+However, for the sake of brevity, we will not tackle tests here.
+Let's just delete them:
+
+[source]
+----
+$ rm -rf src/test
+----
+
+=== Environment
+
+The easiest way to reliably run PostgreSQL and Elasticsearch in your development environment is probably to use docker.
+
+A docker-compose configuration file is available
+link:{demoRepoUrl}/environment-stack.yml[here].
+It includes a cluster of two Elasticsearch nodes and a PostgreSQL instance.
+
+We can start it like this:
+
+[source]
+----
+$ docker-compose -f environment-stack.yml -p hsearch-quarkus-env up
+----
+
+And stop it, removing all docker volumes, like this:
+
+[source]
+----
+$ docker-compose -f environment-stack.yml -p hsearch-quarkus-env down -v
+----
+
+=== Configuration properties
+
+The configuration of database access will go into Quarkus' main configuration file: `application.properties`:
+
+[source]
+----
+quarkus.ssl.native=false <1>
+
+quarkus.datasource.url=jdbc:postgresql://${POSTGRES_HOST}/${POSTGRES_DB} <2>
+quarkus.datasource.driver=org.postgresql.Driver
+quarkus.datasource.username=${POSTGRES_USER}
+quarkus.datasource.password=${POSTGRES_PASSWORD}
+%dev.POSTGRES_HOST=localhost <3>
+%dev.POSTGRES_DB=hsearch_demo
+%dev.POSTGRES_USER=hsearch_demo
+%dev.POSTGRES_PASSWORD=hsearch_demo
```

Comment:
This is a bit weird.

Reply:
Ok, I'll need a hint. What is weird?

Reply:
As discussed, the fact that I override environment variables from this file. I'll change it.